tests/unit/models/gpflux/test_models.py

```diff
@@ -0,0 +1,298 @@
+        kernel,
+        inducing_variable,
+        num_data=5,
+        num_latent_gps=1,
+        mean_function=gpflow.mean_functions.Zero(),
+    )
+
+    likelihood_layer = gpflux.layers.LikelihoodLayer(gpflow.likelihoods.Gaussian(0.01))
+
+    dgp = DeepGP([keras_layer_1, keras_layer_2, gp_layer], likelihood_layer)
+
+    with pytest.raises(ValueError):
+        DeepGaussianProcess(dgp)
+
+
+def test_dgp_warns_for_whitened_layers() -> None:
+    num_data = 5
+    num_inducing = num_data
+
+    Z = np.linspace(-1, 1, num_inducing).reshape(-1, 1)
+    kernel_1 = gpflow.kernels.SquaredExponential()
+    inducing_variable_1 = gpflow.inducing_variables.InducingPoints(Z.copy())
+    gp_layer_1 = gpflux.layers.GPLayer(
+        kernel_1,
+        inducing_variable_1,
+        num_data=num_data,
+        num_latent_gps=1,
+        whiten=False,
+    )
+
+    kernel_2 = gpflow.kernels.SquaredExponential()
+    inducing_variable_2 = gpflow.inducing_variables.InducingPoints(Z.copy())
+    gp_layer_2 = gpflux.layers.GPLayer(
+        kernel_2,
+        inducing_variable_2,
+        num_data=num_data,
+        num_latent_gps=1,
+        whiten=True,
+    )
+
+    dgp = DeepGP([gp_layer_1, gp_layer_2], gpflow.likelihoods.Gaussian(0.01))
+
+    with pytest.warns(UserWarning):
+        DeepGaussianProcess(dgp)
+
+
+def test_dgp_model_attribute() -> None:
+    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
+    dgp = two_layer_dgp_model(x)
+    model = DeepGaussianProcess(dgp)
+
+    assert model.model is dgp
+
+
+def test_dgp_update() -> None:
+    x = tf.zeros([1, 4])
+    dgp = two_layer_dgp_model(x)
+    model = DeepGaussianProcess(dgp)
+
+    assert model.model.num_data == 1
+
+    for layer in model.model.f_layers:
+        assert layer.num_data == 1
+
+    model.update(Dataset(tf.zeros([5, 4]), tf.zeros([5, 1])))
+
+    assert model.model.num_data == 5
+
+    for layer in model.model.f_layers:
+        assert layer.num_data == 5
+
+
+@pytest.mark.parametrize(
+    "new_data",
+    [Dataset(tf.zeros([3, 5]), tf.zeros([3, 1])), Dataset(tf.zeros([3, 4]), tf.zeros([3, 2]))],
+)
+def test_dgp_update_raises_for_invalid_shapes(new_data: Dataset) -> None:
+    x = tf.zeros([1, 4])
+    dgp = two_layer_dgp_model(x)
+    model = DeepGaussianProcess(dgp)
+
+    with pytest.raises(ValueError):
+        model.update(new_data)
+
+
+def test_dgp_optimize_with_defaults() -> None:
+    x_observed = np.linspace(0, 100, 100).reshape((-1, 1))
+    y_observed = fnc_2sin_x_over_3(x_observed)
+    data = x_observed, y_observed
+    dataset = Dataset(*data)
+    optimizer = create_optimizer(tf.optimizers.Adam(), dict(max_iter=20))
+    model = DeepGaussianProcess(two_layer_dgp_model(x_observed), optimizer=optimizer)
+    elbo = model.model.elbo(data)
+    model.optimize(dataset)
+    assert model.model.elbo(data) > elbo
+
+
+def _batcher_1(dataset: Dataset, batch_size: int) -> Iterable[tuple[TensorType, TensorType]]:
```

Comment:
we have these batchers in gpflow tests as well, right? lets move them to a file in `tests/util/models/` folder and load them from there for both gpflow and gpflux

Reply:
or perhaps even better, as a fixture in `tests/unit/models/conftest.py`

Reply:
Should be done now